sw-services/src/main/java/org/egov/swservice/service/EnrichmentService.java

```diff
@@ -0,0 +1,267 @@
+					BigDecimal big = new BigDecimal(String.valueOf(addDetail.get(constKey)));
+					additionalDetail.put(constKey, big);
+				} else {
+					additionalDetail.put(constKey, addDetail.get(constKey));
+				}
+			}
+		}
+		sewerageConnectionRequest.getSewerageConnection().setAdditionalDetails(additionalDetail);
+	}
+	
+	
+	/**
+	 * Sets status for create request
+	 * 
+	 * @param ConnectionRequest
+	 *            The create request
+	 */
+	private void setStatusForCreate(SewerageConnectionRequest sewerageConnectionRequest) {
+		if (sewerageConnectionRequest.getSewerageConnection().getProcessInstance().getAction()
+				.equalsIgnoreCase(SWConstants.ACTION_INITIATE)) {
+			sewerageConnectionRequest.getSewerageConnection().setApplicationStatus(ApplicationStatusEnum.INITIATED);
+		}
+		if (sewerageConnectionRequest.getSewerageConnection().getProcessInstance().getAction().equalsIgnoreCase(SWConstants.ACTION_APPLY)) {
+			sewerageConnectionRequest.getSewerageConnection().setApplicationStatus(ApplicationStatusEnum.APPLIED);
+		}
+	}
+	
+
+
+	/**
+	 * Sets the SewarageConnectionId for given SewerageConnectionRequest
+	 *
+	 * @param request SewerageConnectionRequest which is to be created
+	 */
+	private void setSewarageApplicationIdgenIds(SewerageConnectionRequest request) {
+		List<String> applicationNumbers = getIdList(request.getRequestInfo(), 
+				request.getRequestInfo().getUserInfo().getTenantId(), 
+				config.getSewerageApplicationIdGenName(),
+				config.getSewerageApplicationIdGenFormat(), 1);
+
+		if (CollectionUtils.isEmpty(applicationNumbers) || applicationNumbers.size() != 1) {
+			Map<String, String> errorMap = new HashMap<>();
+			errorMap.put("IDGEN ERROR ",
+					"The Id of SewerageConnection returned by idgen is not equal to number of SewerageConnection");
+			throw new CustomException(errorMap);
+		}
+		request.getSewerageConnection().setApplicationNo(applicationNumbers.listIterator().next());
+	}
+
+	private List<String> getIdList(RequestInfo requestInfo, String tenantId, String idKey, String idformat, int count) {
+		List<IdResponse> idResponses = idGenRepository.getId(requestInfo, tenantId, idKey, idformat, count)
+				.getIdResponses();
+
+		if (CollectionUtils.isEmpty(idResponses))
+			throw new CustomException("IDGEN ERROR", "No ids returned from idgen Service");
+
+		return idResponses.stream().map(IdResponse::getId).collect(Collectors.toList());
+	}
+	
+	/**
+	 * Enrich update sewarage connection
+	 * 
+	 * @param sewarageConnectionRequest
+	 */
+	public void enrichUpdateSewerageConnection(SewerageConnectionRequest sewerageConnectionRequest) {
+		validateProperty.enrichPropertyForSewerageConnection(sewerageConnectionRequest);
+		AuditDetails auditDetails = sewerageServicesUtil
+				.getAuditDetails(sewerageConnectionRequest.getRequestInfo().getUserInfo().getUuid(), false);
+		SewerageConnection connection = sewerageConnectionRequest.getSewerageConnection();
+		if (!CollectionUtils.isEmpty(connection.getDocuments())) {
```

Comment:
uuid should be enriched only if the application is in a state in workflow which has isStateUpdatable = true